core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -474,30 +477,31 @@ private SnapshotInfo inProgressSnapshot(SnapshotsInProgress.Entry entry) {
      * returns similar information but for already finished snapshots.
      * </p>
      *
-     * @param snapshotId snapshot id
+     * @param snapshot snapshot
      * @return map of shard id to snapshot status
      */
-    public Map<ShardId, IndexShardSnapshotStatus> snapshotShards(SnapshotId snapshotId) throws IOException {
-        validate(snapshotId);
+    public Map<ShardId, IndexShardSnapshotStatus> snapshotShards(Snapshot snapshot) throws IOException {
```

Comment:
Pass SnapshotInfo directly to this method? See `TransportSnapshotsStatusAction`. No need to revalidate...